modules/programs/mbsync.nix

```diff
@@ -65,9 +65,11 @@ let
     + genSection "IMAPStore ${name}-remote"
     ({ Account = name; } // mbsync.extraConfig.remote) + "\n"
     + genSection "MaildirStore ${name}-local" ({
-      Path = "${maildir.absPath}/";
       Inbox = "${maildir.absPath}/${folders.inbox}";
-      SubFolders = "Verbatim";
+    } // optionalAttrs (mbsync.subFolders != "Maildir++") {
+      Path = "${maildir.absPath}/";
+    } // optionalAttrs (mbsync.flatten == null) {
+      SubFolders = mbsync.subFolders;
```

Comment:
What happens if `subFolders == "Maildir++" && flatten != null`?

Reply:
The mbsync manual says that the `SubFolders` option will be ignored when `Flatten` is set.

Reply:
I tried setting this in my config and `mbsync` complains about the `Path` not being set.  I should probably add ` || mbsync.flatten != null` to the predicate for `Path`.

Reply:
The folder structure now looks like this

```
maildir
├── Archive
├── cur
├── Drafts
├── folder.sub.subsub
├── Junk
├── new
├── Sent
└── tmp
```

Where `cur`, `new`, and `tmp` is the Inbox